### PR TITLE
docs(refactor): updates to the oauth doc following review

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Review/refactor of the OAUth doc content to make more active and make setup requirements prominent rather than concepts

- Extracted and condensed authorization server setup into single preliminary section
- OAuth authn
    - Broker and client config split (as with the preceding security section)
    - Reduces to only one example for fast-token and introspection endpoint configuration. We had two in different sections
    - Config examples extracted from example procedures and placed in listener and client config sections 
    - Removes the server examples and authentication mechanisms sections to place content in config sections
     - Removes authorization server examples section and adds link to oauth project
- OAuth authorization
    - Streamlines intro and removes intro/overview file
    - Removes the Kafka resources lists (out-of-date and replicated in Kafka docs)
    -  Some minor updates to Keycloak example

**New structure**
image

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/oauth/con-kafka-keycloak-authz-models.adoc
Comment (scholzj) [inline documentation/modules/oauth/con-kafka-keycloak-authz-models.adoc:26]: Do we have these lists somewhere else? They seem to be useful.

File: documentation/assemblies/oauth/assembly-oauth-authorization.adoc
Comment (ppatierno) [inline documentation/assemblies/oauth/assembly-oauth-authorization.adoc:16]: move this on the same line as the previous one?

**Review by ppatierno — APPROVED** [review]
Just a nit but overall it lgtm. Anyway it needs the approval of Marko as SME before merging this.

File: documentation/modules/oauth/con-kafka-keycloak-authz-models.adoc
Comment (PaulRMellor) [reply documentation/modules/oauth/con-kafka-keycloak-authz-models.adoc:26]: We had an out-of-date list. Rather than continuing to maintain it, I thought to add the following with an example and link to the full list in the Kafka docs:

> .Kafka authorization model
> When a Kafka client performs an action on a broker, the broker uses the configured `KeycloakAuthorizer` to check the client's permissions, based on the action and resource type.
> 
> Each resource type has a set of available permissions.
> For example, the `Topic` resource type has `Create` and `Write` permissions among others.
> 
> Refer to the https://kafka.apache.org/documentation/#security_authz_primitives[Kafka authorization model] in the Kafka documentation for the full list of resources and permissions.

**Review by scholzj — APPROVED** [review]
LGTM. Should be reviewed by @mstruk as the SME.

File: documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc
Comment (mstruk) [inline documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc:24]: While not part of this set of changes this sentence is not correct. The way it reads it seems like the authorizer needs to know about Keycloak Authorization Services policies and permissions in order to retrieve grants. But actually, authorizer just sends the access token in an HTTP request, and gets back a list of grants (Kafka ACL permissions in the form of a string of a special format). The evaluation was performed on the Keycloak by Keycloak Authorization Services to produce that list of grants from policies and permission configured for the owner of the access token.

```suggestion
To grant permissions the authorizer evaluates the received (and locally cached) list of grants produced by Keycloak Authorization Services based on policies and permissions configured for the owner of the access token.
```

File: documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc
Comment (mstruk) [inline documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc:46]: These are not supported. We don't support authentication tokens for authentication and we don't support `All` as a valid operation.
```suggestion
```

File: documentation/modules/oauth/con-oauth-authentication-broker.adoc
Comment (mstruk) [inline documentation/modules/oauth/con-oauth-authentication-broker.adoc:51]: The `obtained manually` is no longer the only option as the access token can be made available by some external mechanism and mounted into a pod as a file. An example is the Service Account token file when using Kubernetes API server as authorization server.
```suggestion
* Long-lived access token
```

**Review by mstruk — COMMENTED** [review]
Added some suggestions to improve the documentation.

File: documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc
Comment (mstruk) [reply documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc:46]: For more on delegation tokens see: https://kafka.apache.org/documentation/#security_delegation_token

Comment (PaulRMellor) [reply documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc:46]: Thanks @mstruk : we say this before introducing the list: "Authorization scopes must contain all the possible Kafka permissions regardless of the resource type:"

So, not quite , then?  We should remove the above and say something like: " "Authorization scopes _should_ contain the following Kafka permissions regardless of the resource type:"?

Comment (PaulRMellor) [reply documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc:46]: Should we include a note on delegation tokens or is it implicit?

**Comment by PaulRMellor** [comment]
> Added some suggestions to improve the documentation.

Thanks @mstruk -- making the examples generic is much better

File: documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc
Comment (mstruk) [reply documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc]: The `CreateTokens` and `DescribeTokens` are redundant but by themself not harmful. Adding `All` can lead one to believe that it is a legit operation name to use in order to match any other, which is not supported.
So yes, it would be good to mention at leas the `All`, that using it is not supported.

Comment (PaulRMellor) [reply documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc]: Thanks @mstruk. I've added a note. If you're happy with the changes, please approve.

**Review by mstruk — APPROVED** [review]

**Pull request merged**
